sql/catalyst/src/main/scala/org/apache/spark/sql/sources/aggregates.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.sources
+
+import org.apache.spark.sql.types.DataType
+
+// Aggregate Functions in SQL statement.
+// e.g. SELECT COUNT(EmployeeID), Max(salary), deptID FROM dept GROUP BY deptID
+// aggregateExpressions are (COUNT(EmployeeID), Max(salary)), groupByColumns are (deptID)
+case class Aggregation(aggregateExpressions: Seq[Seq[AggregateFunc]],
+                       groupByColumns: Seq[String])
+
+abstract class AggregateFunc
+
+case class Min(column: String, dataType: DataType) extends AggregateFunc
+case class Max(column: String, dataType: DataType) extends AggregateFunc
```

Comment:
how about `max(a + b)`?

Reply:
This is not supported by Parquet and ORC because they don't have these kind of statistics info. For JDBC,  I treat this as one Column. I have something like this: 
```
  private def columnAsString(e: Expression): String = e match {
    case AttributeReference(name, _, _, _) => name
    case Cast(child, _, _) => columnAsString (child)
    case Add(left, right, _) =>
      columnAsString(left) + " + " + columnAsString(right)
    case Subtract(left, right, _) =>
      columnAsString(left) + " - " + columnAsString(right)
    case Multiply(left, right, _) =>
      columnAsString(left) + " * " + columnAsString(right)
    case Divide(left, right, _) =>
      columnAsString(left) + " / " + columnAsString(right)
    case CheckOverflow(child, _, _) => columnAsString (child)
    case PromotePrecision(child) => columnAsString (child)
    case _ => ""
  }
```
When we do sql("SELECT Max(a+b) FROM t").show, it has:
```
+------------+
|max((a + b))|
+------------+
+------------+
```
I guess it makes sense to treat this as one column?

Reply:
It looks fragile to allow `column: String` to also represent an expression. I think it's better to make it clear in the API and use `org.apache.spark.sql.connector.expressions.Expression`